File: fixtures/scale/python/large/cipherlab/checks/check_nibbleswap.py
from cipherlab.ciphers.nibbleswap import nibbleswap_encode, nibbleswap_decode
from cipherlab.specs.nibbleswap_spec import nibbleswap_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_nibbleswap():
    spec = nibbleswap_spec()
    for text in corpus_codec():
        packed = nibbleswap_encode(text)
        if nibbleswap_decode(packed) != text:
            return False
    return spec["category"] == "codec"

File: fixtures/scale/ruby/large/lib/check_nibbleswap.rb
require_relative 'nibbleswap_encode'
require_relative 'nibbleswap_decode'
require_relative 'corpus_codec'

def check_nibbleswap
  corpus_codec.each do |text|
    packed = nibbleswap_encode(text)
    return false if nibbleswap_decode(packed) != text
  end
  true
end
